core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2227,7 +2227,7 @@ public AlluxioURI getPath(long fileId) throws FileDoesNotExistException {
 
   @Override
   public Set<Long> getPinIdList() {
-    return mInodeTree.getPinIdSet();
+    return Sets.union(mInodeTree.getPinIdSet(), mInodeTree.getToBePersistedIds());
```

Comment:
nit: 
1. can we rename the method or update the comments in `BlockMetadataManagerView:isBlockEvictable` to make it clear that we the evictable check checks for both pinned and to_be_persisted state?
2. ideally the check for free and evict should be the same? in free we can only free PERSISTED files. in evict we allow evicting NOT_PERSISTED files.

✔ ~/os-fork/alluxio [cheng-chang-config-prop-test L|✚ 2]
10:35 $ afs -Dalluxio.user.file.writetype.default=MUST_CACHE copyFromLocal LICENSE /
Copied file:///Users/adit.madan/os-fork/alluxio/LICENSE to /
✔ ~/os-fork/alluxio [cheng-chang-config-prop-test L|✚ 2]
-rw-r--r--  adit.madan     staff                    27040   NOT_PERSISTED 05-10-2019 10:35:44:578 100% /LICENSE
✔ ~/os-fork/alluxio [cheng-chang-config-prop-test L|✚ 2]
10:35 $ afs free /LICENSE
Cannot free file /LICENSE which is not persisted

Reply:
Updated, thanks.